### PR TITLE
Improve the connection error handling code based on suggestion from c…

### PR DESCRIPTION
**Pull request by mbasset**
## Tracker Ticket

* https://quandl.atlassian.net/browse/AP-1657

## Description

* Improvements based on https://github.com/quandl/quandl-python/issues/39

## Is there a migration? 

* No

## Dependency

* None

### CODE REVIEW

**Comment by ccleung** [comment]
:+1:

File: quandl/connection.py [hunk 1]
```diff
@@ -65,21 +65,17 @@ def handle_api_error(cls, resp):
...
+            'L': LimitExceededError,
+            'M': InternalServerError,
+            'A': AuthenticationError,
+            'P': ForbiddenError,
+            'S': InvalidRequestError,
+            'C': NotFoundError,
+            'X': ServiceUnavailableError
+        }
+        klass = QuandlError
+        if code_letter in d_klass.keys():
```
Comment (ccleung) [inline quandl/connection.py]: you don't need `keys()` here

Comment (ccleung) [inline quandl/connection.py]: also we might be able to get rid of this `if` statement, and use this instead:
http://stackoverflow.com/questions/9358983/python-dictionary-and-default-values

Dictionaries can accept default values. Just a suggestion, no need to change if we prefer the current PR's style

Comment (mbasset) [inline quandl/connection.py]: Will try that alternative method.

**Comment by ccleung** [comment]
:+1:

**Pull request merged**